### DISCUSSION TITLE
Last step turns to black and white

### DISCUSSION THREAD

**Discussion by hofnik** [Mikubill/sd-webui-controlnet#120] [General]
I see all the steps being generated in colors but the very last and the saved generated image is always black and white, in any models.
Any ideas?

**Comment by Mikubill** [comment]
Any screenshots? Note that some type of detectmap always black and white.

**Comment by hofnik** [comment]
I restarted the whole thing and now to the very last step shifts the color a bit, as you can see here, I have no idea what's this...

![image](https://user-images.githubusercontent.com/68555622/219524421-e5ae7414-7be6-4134-94f2-e837e7451e5b.png)

**Reply by hofnik** [reply]
![image](https://user-images.githubusercontent.com/68555622/219543942-5299719c-c89c-4721-a44f-66245c132a0e.png)

**Comment by hofnik** [comment]
![00176-1575336784-photo of girl with a red hat, rainbow colors](https://user-images.githubusercontent.com/68555622/219686181-123268eb-c789-4d11-bc1f-0811596dd13c.png)

https://user-images.githubusercontent.com/68555622/219686253-fad24469-2192-453d-962f-fa5c53dbc3a9.mp4

All the steps have colors (video), and the rendered final image is monochrome...

**Comment by catboxanon** [comment]
You're using img2img and you probably also have color correction for that mode enabled. This extension is moreso meant for txt2img, img2img support was only added because it just works but it's not inherently meant for that mode.

**Reply by hofnik** [reply]
Oh it was on img2img, you're right!
Where is this color correction option again?
Thanks so much for this!

**Reply by catboxanon** [reply]
It's under `Stable Diffusion`. You may or may not have it enabled, I was only assuming it is based on the colors vastly changing at the last step. It's a fine option to have enabled for normal img2img usage.
